src/containers/Routes.js

```diff
@@ -18,10 +19,11 @@ class Routes extends Component {
                     <Route exact path="/">
                         <Redirect to="home"/>
                     </Route>
+                    <Route path="/signin" component={Signin}/>
+                    <Route path="/signup" component={Signup}/>
+                    <Route path="/products/:slug" component={ProductCategory}/>
+                    <Route path="/detail/:slug" component={ProductDetail}/>
```

Comment:
Route detail should be: `/products/:productId/detail` or just simply `/products/:productId`.
https://blog.mwaysolutions.com/2014/06/05/10-best-practices-for-better-restful-api/